libtenzir/src/data.cpp

```diff
@@ -445,7 +445,7 @@ record flatten(const record& r, size_t max_recursion) {
   }
   for (const auto& [k, v] : r) {
     if (const auto* nested = try_as<record>(&v)) {
-      for (auto& [nk, nv] : flatten(*nested, --max_recursion)) {
+      for (auto& [nk, nv] : flatten(*nested, max_recursion - 1)) {
```

Comment:
The two functions below that have exactly the same bug. Can you fix them as well?

Reply:
Good spot. Pushed that as well.